### ISSUE TITLE
[JS API] `.set("imports", [])` overrides internally required `imports` option

### ISSUE THREAD

**Issue by qnp** [stylus/stylus#2585]
Greetings,

## Obsevations

As I coined in *Vite* project (https://github.com/vitejs/vite/issues/2857), compiling stylus source with the following code will raise an error because `.set` completely overrides stylus option that are required for correct compilation, such as `imports: ['node_modules/stylus/lib/functions/index.styl']`:

```js
const stylus = require('stylus')
const source = `$easing := cubic-bezier(0.4, 0.0, 0.2, 1)`
const options = { imports: ['path/to/my/variables.styl'] }

const ref = stylus(source)
// Here if you console.log(ref), you will see that this ref has
// an `import` option with the required internal file I mentioned

Object.keys(options).forEach((key) => ref.set(key, options[key]))
// Fails with "Cannot read property 'nodes' of undefined"
const result = ref.render() 
```

## What is expected
` ref.set(...)` should *merge* options properly.

Workarounds are still possible (that is to say, pass options directly to `const ref = stylus(source, options)`, but I had to mention this buggy behgaviour.

Best regards,

**Comment by zouhangwithsweet** [comment]
I have the same problem

**Comment by iChenLei** [comment]
Yes, this case is cause by design. tj design some built-in functions via `functions/index.styl`. So PRs welcome.
```js
  options.imports = [join(__dirname, 'functions/index.styl')].concat(options.imports || []);
```
https://github.com/stylus/stylus/blob/baba86d07c1c4c1719873f1f5893d865d062c8a7/lib/renderer.js#L162-L164
